Formula/dafny.rb

```diff
@@ -0,0 +1,28 @@
+class Dafny < Formula
+  desc "Verification-aware programming language"
+  homepage "https://www.microsoft.com/en-us/research/project/dafny-a-language-and-program-verifier-for-functional-correctness"
+  url "https://github.com/dafny-lang/dafny/releases/download/v2.3.0/dafny-2.3.0.10506-x64-osx-10.14.1.zip",
```

Comment:
This looks platform specific. That's not allowed, all formula need to build from source.

Reply:
@SMillerDev  The original package was built with z3 which is platform specific. So I removed it during the installation, and make it as a dependency since there is a existed formula. If you look the [build process](https://github.com/dafny-lang/dafny/blob/master/INSTALL.md#building-on-mac-os-x), you will find z3 is the only platform specific.

Reply:
Yeah, I still object to platform specific downloads. If not all of it is cross platform it should build from source.

Reply:
@SMillerDev I just changed the file to make it built from source. No platform specific downloads now.